### PR TITLE
Oracle Developer Studio 12.6 fixes

### PR DESCRIPTION
**Pull request by k0ekk0ek**
The compiler has problems with `getopt` (not technically necessary for NSD), so the configure checks need to be extended a little bit. Apart from that, the functions need unit tests to verify they do exactly as required because and to verify they don't introduce extra work. Work-in-progress.

### CODE REVIEW

**Comment by k0ekk0ek** [comment]
Compile under Oracle Developer Studio 12.6 still fails because `tempnam` is missing. It's used only in tests, but we needed to use another function for that because of warnings anyway, so fixing that next. Still need to add tests to verify correct operation of lower level functions like `trailing_zeroes` etc.

**Comment by k0ekk0ek** [comment]
Well it builds and runs if I compile using Oracle Developer Studio 12.6. Now to add tests etc.

**Comment by k0ekk0ek** [comment]
Still some warnings when compiling with developer studio, but builds and tests all run.

**Comment by k0ekk0ek** [comment]
Changes so far allow me to build for Solaris 10 (sparce and x86_64). Now going to test building NSD 4.10.

**Comment by k0ekk0ek** [comment]
Once NLnetLabs/nsd#351 is merged, simdzone compiles just fine. At least, with GCC. simdzone builds just fine with Developer Studio 12.4 too, but NSD itself does not. Verified for x86 and Sparc using build infrastructure from https://www.opencsw.org/.

See log: 
[solaris.build.log](https://github.com/user-attachments/files/16191438/solaris.build.log)

**Comment by k0ekk0ek** [comment]
Quick push to resolve merge issues. Should be all good now.

File: tests/bounds.c [hunk 1]
```diff
@@ -228,7 +229,7 @@ void contiguous_on_buffer_boundary(void **state)
   zone_buffers_t buffers = { 1, &owner, &rdata };
 
   // generate zone file to parse
-  char *path = tempnam(NULL, "xbounds");
+  char *path = get_tempnam(NULL, "xbounds");
   assert_non_null(path);
   FILE *handle = fopen(path, "wb");
```
Comment (wcawijngaards) [inline tests/bounds.c:234]: The `path` is not unlinked, with like `remove_include(path);` or `unlink(path);` before `free(path);` at the end of the function.

File: tests/fallback/bits.c [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    uint64_t tz = trailing_zeroes(bit);
+    assert_int_equal(tz, shift);
+  }
+}
+
+void test_fallback_leading_zeroes(void **state)
+{
+  (void)state;
+  fprintf(stderr, "test_fallback_leading_zeroes\n");
+  for (uint64_t shift = 0; shift << 63; shift++) {
```
Comment (wcawijngaards) [inline tests/fallback/bits.c:34]: The condition was not a comparison.
```suggestion
  for (uint64_t shift = 0; shift < 63; shift++) {
```

File: tests/haswell/bits.c [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    uint64_t tz = trailing_zeroes(bit);
+    assert_int_equal(tz, shift);
+  }
+}
+
+void test_haswell_leading_zeroes(void **state)
+{
+  (void)state;
+  fprintf(stderr, "test_haswell_leading_zeroes\n");
+  for (uint64_t shift = 0; shift << 63; shift++) {
```
Comment (wcawijngaards) [inline tests/haswell/bits.c:33]: The condition of the for was wrong.
```suggestion
  for (uint64_t shift = 0; shift < 63; shift++) {
```

File: tests/westmere/bits.c [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    uint64_t tz = trailing_zeroes(bit);
+    assert_int_equal(tz, shift);
+  }
+}
+
+void test_westmere_leading_zeroes(void **state)
+{
+  (void)state;
+  fprintf(stderr, "test_westmere_leading_zeroes\n");
+  for (uint64_t shift = 0; shift << 63; shift++) {
```
Comment (wcawijngaards) [inline tests/westmere/bits.c:33]: The condition was not a comparison.
```suggestion
  for (uint64_t shift = 0; shift < 63; shift++) {
```

**Review by wcawijngaards — APPROVED** [review]
Many changes, but it is nice to increase the portability. There are added tests for the compat simd implementations.

File: tests/westmere/bits.c (hunk 4 above)
Comment (k0ekk0ek) [reply tests/westmere/bits.c]: fixed

File: tests/haswell/bits.c (hunk 3 above)
Comment (k0ekk0ek) [reply tests/haswell/bits.c]: fixed

File: tests/fallback/bits.c (hunk 2 above)
Comment (k0ekk0ek) [reply tests/fallback/bits.c]: fixed

File: tests/bounds.c (hunk 1 above)
Comment (k0ekk0ek) [reply tests/bounds.c:235]: fixed

**Pull request merged**